src/integrations/windowsIntegrationManager.ts

```diff
@@ -1,14 +1,430 @@
+  'docker-desktop-data', // Not meant for interactive use
+];
 
 /**
- * TODO: change this comment once Windows integration management is refactored.
- * Doesn't do anything at the moment. Integration management on Windows
- * still is done inside the WSLBackend object. This class exists because
- * we need an IntegrationManager for Windows that does nothing until we can
- * write one that actually manages integrations.
+ * WindowsIntegrationManager manages various integrations on Windows, for both
+ * the Win32 host, as well as for each (foreign) WSL distribution.
+ * This includes:
+ * - Docker socket forwarding.
+ * - Kubeconfig.
+ * - Docker-compose executable (WSL distributions only).
  */
 export default class WindowsIntegrationManager implements IntegrationManager {
-  async enforce(): Promise<void> {}
-  async remove(): Promise<void> {}
+  /** A snapshot of the application-wide settings. */
+  protected settings: RecursivePartial<Settings> = {};
+
+  /** Background processes for docker socket forwarding, per WSL distribution. */
+  protected distroSocketProxyProcesses: Record<string, BackgroundProcess> = {};
+
+  /** Background process for docker socket forwarding to the Windows host. */
+  protected windowsSocketProxyProcess: BackgroundProcess;
+
+  /** Whether integrations as a whole are enabled. */
+  protected enforcing = false;
+
+  /** Whether the backend is in a state where the processes should run. */
+  protected backendReady = false;
+
+  /** Extra debugging arguments for wsl-helper. */
+  protected wslHelperDebugArgs: string[] = [];
+
+  constructor() {
+    mainEvents.on('settings-update', (settings) => {
+      this.wslHelperDebugArgs = settings.debug ? ['--verbose'] : [];
+      this.settings = JSON.parse(JSON.stringify(settings));
+      this.sync();
+    });
+    mainEvents.on('k8s-check-state', (mgr) => {
+      this.backendReady = [State.STARTED, State.STARTING, State.DISABLED].includes(mgr.state);
+      this.sync();
+    });
+    this.windowsSocketProxyProcess = new BackgroundProcess(
+      'Win32 socket proxy', async() => {
+        const stream = await Logging['wsl-helper'].fdStream;
+
+        console.debug('Spawning Windows docker proxy');
+
+        return spawn(
+          path.join(paths.resources, 'win32', 'wsl-helper.exe'),
+          ['docker-proxy', 'serve', ...this.wslHelperDebugArgs], {
+            stdio:       ['ignore', stream, stream],
+            windowsHide: true,
+          });
+      });
+
+    // Trigger a settings-update.
+    mainEvents.emit('settings-write', {});
+  }
+
+  async enforce(): Promise<void> {
+    this.enforcing = true;
+    await this.sync();
+  }
+
+  async remove(): Promise<void> {
+    this.enforcing = false;
+    await this.sync();
+  }
+
+  async sync(): Promise<void> {
+    try {
+      await Promise.all([
+        this.syncSocketProxy(),
+        this.syncDockerCompose(),
+        this.syncKubeconfig(),
+      ]);
+    } finally {
+      await this.listIntegrations();
```

Comment:
This is surprising to me as a reader. It isn't clear why this is here without looking at `listIntegrations`. I would do `mainEvents.emit('integration-update', await this.listIntegrations())` here and remove the `integration-update` trigger from `listIntegrations`.